adam/language_specific/chinese/chinese_language_generator.py

```diff
@@ -360,12 +360,21 @@ def _collect_action_modifiers(
                         path
                         for (_, path) in action.during.objects_to_paths.items()
                         if (
-                            isinstance(path.reference_object, SituationObject)
-                            and path.reference_object.ontology_node == GROUND
+                            isinstance(path.reference_source_object, SituationObject)
```

Comment:
nit: maybe add a one-line comment here explaining what you're checking for on a high level